jvm-agent/src/main/org/jetbrains/lincheck/jvm/agent/analysis/controlflow/LoopInformation.kt

```diff
@@ -266,14 +290,21 @@ internal fun BasicBlockControlFlowGraph.computeLoopsFromDominators(): MethodLoop
                 }
             }
         }
+        // All exits that are only reachable from the loop body
+        val exclusiveExits = sequence {
```

Comment:
Why you decided to use `sequence` here, is there any reason?
In general, `sequence` builder is not cheap, unless you really need lazy elements processing better to use regular lists.

Reply:
Alternatively, I could have something like this:
```kotlin
val exclusiveExits = mutableSetOf<BasicBlockIndex>().apply {
  normalExits.map { it.target }.filterTo(this) { isExclusiveExitFromLoopBody(it, body) }
  exceptionalExitHandlers.filterTo(this) { isExclusiveExitFromLoopBody(it, body) }
}
```

I think sequence variant is cleaner

Reply:
- https://chrisbanes.me/posts/use-sequence/
- https://youtrack.jetbrains.com/issue/KT-75201